### DISCUSSION TITLE
fn function error

### DISCUSSION THREAD

**Discussion by survuvi** [modular/modular#169] [Q&A] [answered]
What is wrong? With def instead of fn it works properly

```python

fn main():
    for i in range(1, 11):
        print(i * 5)
main()

```

_error: Expression [10]:19:9: ambiguous call to 'main', each candidate requires 0 implicit conversions, disambiguate with an explicit cast
    main()
  
Expression [10]:5:1: candidate declared here
fn main():

Expression [9]:5:1: candidate declared here
def main():_

**Comment by lattner** [comment, marked as answer]
This is because the workbook is wrapping the top level code in a main() function, which yours conflicts with.  You can workaround by choosing a different name.

@River707 Do you have any ideas on how to approach this or improve the error message?

**Reply by survuvi** [reply]
Yes now it works, thank you
